cockatrice/src/spoilerbackgroundupdater.cpp

```diff
@@ -0,0 +1,210 @@
+#include "window_main.h"
+
+#define SPOILERS_STATUS_URL "https://raw.githubusercontent.com/Cockatrice/Magic-Spoiler/files/SpoilerSeasonEnabled"
+#define SPOILERS_URL "https://raw.githubusercontent.com/Cockatrice/Magic-Spoiler/files/spoiler.xml"
+
+SpoilerBackgroundUpdater::SpoilerBackgroundUpdater(QObject *apParent) : QObject(apParent), cardUpdateProcess(nullptr)
+{
+    isSpoilerDownloadEnabled = settingsCache->getDownloadSpoilersStatus();
+    if (isSpoilerDownloadEnabled)
+    {
+        // Start the process of checking if we're in spoiler season
+        // File exists means we're in spoiler season
+        startSpoilerDownloadProcess(SPOILERS_STATUS_URL, false);
+    }
+}
+
+void SpoilerBackgroundUpdater::startSpoilerDownloadProcess(QString url, bool saveResults)
+{
+    auto spoilerURL = QUrl(url);
+    downloadFromURL(spoilerURL, saveResults);
+}
+
+void SpoilerBackgroundUpdater::downloadFromURL(QUrl url, bool saveResults)
+{
+    auto *nam = new QNetworkAccessManager(this);
+    QNetworkReply *reply = nam->get(QNetworkRequest(url));
+
+    if (saveResults)
+    {
+        // This will write out to the file (used for spoiler.xml)
+        connect(reply, SIGNAL(finished()), this, SLOT(actDownloadFinishedSpoilersFile()));
+    }
+    else
+    {
+        // This will check the status (used to see if we're in spoiler season or not)
+        connect(reply, SIGNAL(finished()), this, SLOT(actCheckIfSpoilerSeasonEnabled()));
+    }
+}
+
+void SpoilerBackgroundUpdater::actDownloadFinishedSpoilersFile()
+{
+    // Check for server reply
+    auto *reply = dynamic_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError errorCode = reply->error();
+
+    if (errorCode == QNetworkReply::NoError)
+    {
+        spoilerData = reply->readAll();
+        reply->deleteLater();
+
+        // Save the spoiler.xml file to the disk
+        saveDownloadedFile(spoilerData);
+    }
+    else
+    {
+        qDebug() << "Error downloading spoilers file" << errorCode;
+    }
+}
+
+void SpoilerBackgroundUpdater::actCheckIfSpoilerSeasonEnabled()
+{
+    auto *response = dynamic_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError errorCode = response->error();
+
+    if (errorCode == QNetworkReply::ContentNotFoundError)
+    {
+        // Spoiler season is offline at this point, so the spoiler.xml file can be safely deleted
+        // The user should run Oracle to get the latest card information
+        QString fileName = settingsCache->getSpoilerCardDatabasePath();
+        QFileInfo fi(fileName);
+        QDir fileDir(fi.path());
+        QFile file(fileName);
+
+        // Delete the spoiler.xml file as we're not in spoiler season
+        if (file.exists() && file.remove())
+        {
+            qDebug() << "Spoiler Season Offline, Deleting spoiler.xml";
+        }
+
+        /*
+         * ALERT: Ensure two reloads of the card database do not happen
+         * at the same time or a racetime condition can/will happen!
```

Comment:
We should probably have a mutex around db reloading to avoid that, but maybe in another pr unless you feel up to adding that. Might need refactoring

Reply:
Agreed. I'm going to hold off on that so we can get this PR in and i'll start working on it later.